diffyscan/utils/calldata.py

```diff
@@ -0,0 +1,45 @@
+from .logger import logger
+from .encoder import encode_constructor_arguments
+
+def get_calldata(contract_address_from_config, target_compiled_contract, binary_config):
+    calldata = get_prepared_calldata_from_config(contract_address_from_config, binary_config)
+    if calldata is not None:
+          return calldata, ''
+    
+    calldata, text_error = parse_calldata_from_config(
+        contract_address_from_config,
+        binary_config["constructor_args"],
+        target_compiled_contract,
+    )
+    return calldata, text_error
+            
+def get_constructor_abi(target_compiled_contract):
+    constructor_abi = None
+    try:
+        constructor_abi = [entry["inputs"] for entry in target_compiled_contract['abi'] if entry["type"] == "constructor"][0]
+    except IndexError:
+        logger.okay(f'Constructor in ABI not found, calldata calculation skipped')
+        return None
+
+    logger.okay(f'Constructor in ABI successfully found: {[arg['type'] for arg in constructor_abi]}')
```

Comment:
might be too verbose to log every step

Reply:
reduced to: "Constructor in ABI successfully found"